include/alpaka/core/CallbackThread.hpp

```diff
@@ -51,17 +60,17 @@ namespace alpaka::core
         std::thread m_thread;
         std::condition_variable m_cond;
         std::mutex m_mutex;
-        std::atomic<bool> m_stop{false};
+        bool m_stop{false};
         std::queue<Task> m_tasks;
 
         auto startWorkerThread() -> void
         {
             m_thread = std::thread(
                 [this]
                 {
-                    Task task;
                     while(true)
                     {
+                        Task task;
```

Comment:
I opened a PR  #1976 for this bugfix only to avoid the bugfix is sneaked in with other changes which are an improvement and not a bugfix.

Reply:
Great, I will rebase on your changes!

And in my understanding, improvements is a very general vage term and can include bug fixes :)